zerver/actions/message_edit.py

```diff
@@ -1244,6 +1266,71 @@ def check_time_limit_for_change_all_propagate_mode(
     )
 
 
+def check_time_limit_for_change_all_or_stream_edit_request(
+    message: Message,
+    user_profile: UserProfile,
+    stream_id: int | None,
+    topic_name: str | None,
+    propagate_mode: str,
+    edit_limit_buffer: int,
+    is_resolve_topic_request: bool,
+) -> None:
+    if (
+        stream_id is not None
+        and user_profile.realm.move_messages_between_streams_limit_seconds is not None
+        and not user_profile.is_realm_admin
+        and not user_profile.is_moderator
+    ):
+        deadline_seconds = (
+            user_profile.realm.move_messages_between_streams_limit_seconds + edit_limit_buffer
+        )
+        if (timezone_now() - message.date_sent) > timedelta(seconds=deadline_seconds):
+            raise JsonableError(_("The time limit for editing this message's channel has passed"))
+
+    if (
+        propagate_mode == "change_all"
+        and not is_resolve_topic_request
+        and not user_profile.is_realm_admin
+        and not user_profile.is_moderator
+        and (topic_name is not None or stream_id is not None)
+    ):
+        check_time_limit_for_change_all_propagate_mode(message, user_profile, topic_name, stream_id)
+
+
+def check_message_or_topic_edit_permissions(
```

Comment:
I think this should be called `check_move_messages_within_channel_permissions` -- that at least seems to be the only things it's checking.

But I'm also not sure it's net helpful to have this function exist at all -- maybe the better boundary would be to just have a helper `check_move_messages_time_limit`, and have the first `user_profile.can_move_messages_to_another_topic` check live in the caller.

Reply:
>have a helper check_move_messages_time_limit, and have the first user_profile.can_move_messages_to_another_topic check live in the caller.

This definitely sounds better